packages/http-client-java/emitter/src/code-model-builder.ts

```diff
@@ -992,43 +995,147 @@ export class CodeModelBuilder {
+                return lastSegment;
+              }
             }
+            return undefined;
           }
-          if (nextLinkSegments) {
-            const lastSegment = nextLinkSegments[nextLinkSegments.length - 1];
-            if (lastSegment.kind === "property") {
-              nextLinkSerializedName = getPropertySerializedName(lastSegment);
+          function getLastSegment(
+            segments: SdkModelPropertyType[] | undefined,
+          ): SdkModelPropertyType | undefined {
+            if (segments) {
+              return segments[segments.length - 1];
             }
+            return undefined;
+          }
+          function getLastSegmentSerializedName(
+            segments: SdkModelPropertyType[] | undefined,
+          ): string | undefined {
+            const lastSegment = getLastPropertySegment(segments);
+            return lastSegment ? getPropertySerializedName(lastSegment) : undefined;
           }
 
-          op.responses?.forEach((r) => {
-            if (r instanceof SchemaResponse) {
-              this.trackSchemaUsage(r.schema, { usage: [SchemaContext.Paged] });
+          // TODO: in future the property could be nested, so that the "itemSegments" or "nextLinkSegments" would contain more than 1 element
+          // item/result
+          // "itemsSegments" should exist for "paging"/"lropaging"
+          const itemSerializedName = getLastSegmentSerializedName(
+            sdkMethod.response.resultSegments,
+          );
+          // nextLink
+          const nextLinkSerializedName = getLastSegmentSerializedName(
+            sdkMethod.pagingMetadata.nextLinkSegments,
+          );
+
+          // continuationToken
+          let continuationTokenParameter: Parameter | undefined;
+          let continuationTokenResponseProperty: Property[] | undefined;
+          let continuationTokenResponseHeader: HttpHeader | undefined;
+          if (!this.isBranded()) {
+            const continuationTokenParameterSegment = getLastSegment(
+              sdkMethod.pagingMetadata.continuationTokenParameterSegments,
+            );
+            const continuationTokenResponseSegment = getLastSegment(
+              sdkMethod.pagingMetadata.continuationTokenResponseSegments,
+            );
+            if (continuationTokenParameterSegment) {
+              // for now, continuationToken is either request query or header parameter
+              const parameter = getHttpOperationParameter(
+                sdkMethod,
+                continuationTokenParameterSegment,
+              );
+              if (parameter && op.parameters) {
+                for (const param of op.parameters) {
+                  if (param.protocol.http?.in === parameter.kind) {
+                    if (
+                      parameter.kind === "header" &&
+                      param.language.default.serializedName.toLowerCase() ===
+                        parameter.serializedName.toLowerCase()
```

Comment:
just curious - is "header"'s serialized-name something special needing `toLowerCase` but not for "query"

Reply:
Just defensive coding, as header name should be case insensitive. I am not able to be certain after all the processing (some not in this lib) whether some be "content-type" and some be "Content-Type".